Controller/Connect/Cancel.php

```diff
@@ -95,7 +104,7 @@ public function execute()
 
         $order = $this->checkoutSession->getLastRealOrder()->loadByIncrementId($orderId);
 
-        $order->cancel();
+        $this->orderService->cancel($order->getEntityId());
```

Comment:
The old code will still be used by the order service, but it will also do additional things:
- It will check if an order can be cancelled and if not, it will abort the cancel procedure
- It will cancel the payment like in the old code and trigger event 'order_cancel_after' (exactly what `$order->cancel()` does)
- It will notify the repository to store the changes of the order, so the order state is correct (which is not the case without this patch).

References: 
- cancel method in orderService (saving of order is done with `$this->metadata->getMapper()->save($entity);`) https://github.com/magento/magento2/blob/ed2ffad8f227e253c0aa50393f40478b02310b65/app/code/Magento/Sales/Model/Service/OrderService.php#L107-L112
- cancel method of the order model, which was already used before this patch was applied: https://github.com/magento/magento2/blob/ed2ffad8f227e253c0aa50393f40478b02310b65/app/code/Magento/Sales/Model/Order.php#L1275-L1280